### PR TITLE
added rateus feature in home page

### LINKED ISSUE #2255: [Feature Request]: Add Rate Us page
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

I would like to add Rate us page in this website.

### Use Case

This helps the user to quickly review the website and also suggest a change or improvement by providing feedback.

### Benefits

_No response_

### Add ScreenShots

_No response_

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by MuraliDharan7**
## Related Issue
- Closes #2255 

## Description

Added Rateus feature where users can rate the page out of 5, also can give feedback about the same.

## Type of PR

- [X ] Feature enhancement

## Screenshots / Videos (if applicable)

https://github.com/CodeHarborHub/codeharborhub.github.io/assets/130668745/f65d783e-7423-4a6c-a174-0a3a305f86d5

## Checklist
- [X ] I have performed a self-review of my code.
- [X ] I have read and followed the Contribution Guidelines.
- [X ] I have tested the changes thoroughly before submitting this pull request.
- [X ] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [X ] I have commented my code, particularly in hard-to-understand areas.
- [X ] I have followed the code style guidelines of this project.
- [X ] I have checked for any existing open issues that my pull request may address.
- [ ] I have ensured that my changes do not break any existing functionality.
- [X ] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently.
- [X ] I have read the resources for guidance listed below.
- [ X] I have followed security best practices in my code changes.

### CODE REVIEW

**Comment by MuraliDharan7** [comment]
@Ajay-Dhangar Can you guide through the errors, it would be helpful for me to solve the error as fast.

File: src/components/RateUsPopup/RateUsPopup.module.css [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  
+  .starRating .filled {
+    color: gold;
+  }
+  
+  .starRating .empty {
+    color: lightgray;
+  }
+  
+  textarea {
```
Comment (Ajay-Dhangar) [inline src/components/RateUsPopup/RateUsPopup.module.css:53]: replace `textarea` to `.popupOverlay form textarea` for good practice

File: src/components/Tweet/index.tsx [hunk 2]
```diff
@@ -1,10 +1,9 @@
-import React, {type ReactNode} from 'react';
+import React, {useState, type ReactNode} from 'react';
```
Comment (Ajay-Dhangar) [inline src/components/Tweet/index.tsx:1]: `useState` is define but not use it. If not using then remove for good practice

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by MuraliDharan7** [comment]
Could you please check the changes?

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**